src/resolvers/projectResolver.ts

```diff
@@ -567,221 +573,88 @@ export class ProjectResolver {
     private readonly projectImageRepository: Repository<ProjectImage>,
   ) {}
 
-  // Backward Compatible Projects Query with added pagination, frontend sorts and category search
   @Query(returns => AllProjects)
-  async projects(
+  async allProjects(
     @Args()
     {
-      take,
+      limit,
       skip,
-      orderBy,
       searchTerm,
       category,
       mainCategory,
-      filterBy,
-      admin,
+      filters,
+      sortingBy,
       connectedWalletUserId,
     }: GetProjectsArgs,
-    @Ctx() { req: { user } }: MyContext,
+    @Ctx() { req: { user }, projectsFiltersThreadPool }: MyContext,
   ): Promise<AllProjects> {
-    const categories = await Category.find();
-    let query = this.projectRepository
-      .createQueryBuilder('project')
-      .leftJoinAndSelect('project.status', 'status')
-      .leftJoinAndSelect('project.users', 'users')
-      .leftJoinAndSelect('project.addresses', 'addresses')
-      .leftJoinAndSelect('project.organization', 'organization')
-      // you can alias it as user, but it still is mapped as adminUser
-      // like defined in our project entity
-      .innerJoin('project.adminUser', 'user')
-      .addSelect(publicSelectionFields) // aliased selection
-      .leftJoinAndSelect(
-        'project.categories',
-        'categories',
-        'categories.isActive = :isActive',
-        { isActive: true },
-      )
-      .leftJoinAndSelect('categories.mainCategory', 'mainCategory')
-      .where(
-        `project.statusId = ${ProjStatus.active} AND project.listed = true`,
-      )
-      .leftJoin('project.projectPower', 'projectPower')
-      .addSelect([
-        'projectPower.totalPower',
-        'projectPower.powerRank',
-        'projectPower.round',
-      ]);
-    // Filters
-    query = ProjectResolver.addCategoryQuery(query, category);
-    query = ProjectResolver.addMainCategoryQuery(query, mainCategory);
-    query = ProjectResolver.addSearchQuery(query, searchTerm);
-    query = ProjectResolver.addFilterQuery(
-      query,
-      filterBy?.field,
-      filterBy?.value,
+    const userId = user?.id || 0;
+    const projectsQuery = filterProjectsQuery(
+      limit,
+      skip,
+      searchTerm,
+      category,
+      mainCategory,
+      filters,
+      sortingBy,
+      connectedWalletUserId,
+      user,
```

Comment:
I think we must skip the user from the key. I mean fetch the projects independent of the user, then in a separate query load the user reactions and fill projects! Of course, db does the joining better than us, but 
- For every user cache value is useless
- User may update the reaction (like/unlike a project)! and the cache will not be invalidated immediately

Reply:
As we decided by DM, I will remove the reaction of the logged in user portion from the cache and the userId from the cache key. 
Query the reaction separately and then in a Worker merge the data.